Add Wayback Machine URL archiver and replacer script

kritibirda opened the pull request:
<!--
  Thanks for contributing to holidays!
-->

## Proposed change

<!--
  Describe the big picture of your changes.
  Don't forget to link your PR to an existing issue if any.
-->
Add Wayback Machine URL archiver and replacer script

- Recursively scans the `holidays` package for `.py` and `.po` files, ignoring `__pycache__`
- Extracts all HTTP(S) links, filtering out known domains (e.g., GitHub, Wikipedia, Python docs)
- Queries the Wayback Machine CDX API for existing captures
- Submits URLs to the Wayback Machine Save API when needed
- Supports three archive policies:
  • `if-missing` (default): archive only when no capture exists
  • `always`: always submit for archiving, even if captures exist
  • `never`: only lookup existing captures, do not archive
- Replaces original URLs in source files with their Wayback snapshots
- Uses a retrying `requests` session with exponential backoff for robustness
- Prints progress summaries and warning messages for files that cannot be read or written

This script makes it easy to freeze external references within the codebase, ensuring all links remain valid over time.

Fix https://github.com/vacanza/holidays/issues/2467.

## Type of change

<!--
  Type of change you want to introduce. Please, check one (1) box only!
  If your PR requires multiple boxes to be checked, most likely it needs to
  be split into multiple PRs.
-->

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [X] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

<!--
  Put an `x` in the boxes that apply. You can change them after PR is created.
-->

- [X] I've followed the [contributing guidelines][contributing-guidelines]
- [X] I've successfully run `make check`, all checks and tests are green

<!--
  Thanks again for your contribution!
-->

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.md
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

kritibirda committed:
feat: add Wayback Machine URL archiver and replacer script

This commit introduces a new CLI script that:

- Recursively scans the `holidays` package for `.py` and `.po` files, ignoring `__pycache__`
- Extracts all HTTP(S) links, filtering out known domains (e.g., GitHub, Wikipedia, Python docs)
- Queries the Wayback Machine CDX API for existing captures
- Submits URLs to the Wayback Machine Save API when needed
- Supports three archive policies:
  • `if-missing` (default): archive only when no capture exists
  • `always`: always submit for archiving, even if captures exist
  • `never`: only lookup existing captures, do not archive
- Replaces original URLs in source files with their Wayback snapshots
- Uses a retrying `requests` session with exponential backoff for robustness
- Prints progress summaries and warning messages for files that cannot be read or written

This script makes it easy to freeze external references within the codebase, ensuring all links
remain valid over time.

kritibirda committed:
chore: run the archive links script on the codebase

kritibirda26 commented:
There are still 41 files with ~177 URLs that could not be archived by Internet Archive. Some of these URLs might be no longer accessible.

kritibirda committed:
do not replace link to semver.org

PPsyrius commented on holidays/countries/australia.py:
```diff
@@ -29,24 +29,24 @@ class Australia(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, S
+          * [NSW Public Holidays Act 2010](https://web.archive.org/web/20250316173922/https://legislation.nsw.gov.au/view/html/inforce/current/act-2010-115)
+          * [NT Public Holidays Act 1981](https://web.archive.org/web/20250315072128/https://legislation.nt.gov.au/api/sitecore/Act/PDF?id=12145)
+          * [QLD Holidays Act 1983](https://web.archive.org/web/20250404230918/https://www.legislation.qld.gov.au/view/html/inforce/current/act-1983-018)
+          * [QLD 2012-2024](https://web.archive.org/web/20250414072303/https://web.archive.org/web/20250415220029/https://www.qld.gov.au/recreation/travel/holidays/public)
```
```suggestion
          * [QLD 2013-2028](https://web.archive.org/web/20150703042947/http://www.qld.gov.au/recreation/travel/holidays/public/)
```

PPsyrius commented on holidays/countries/belarus.py:
```diff
@@ -23,16 +23,16 @@ class Belarus(HolidayBase, ChristianHolidays, InternationalHolidays, StaticHolid
 
     Cross-checked With:
-        * <https://president.gov.by/en/gosudarstvo/prazdniki/calendar-2024>
+        * <https://web.archive.org/web/20250415105200/https://web.archive.org/web/20250421002507/https://president.gov.by/en/gosudarstvo/prazdniki/calendar-2024>
```
```suggestion
        * <https://web.archive.org/web/20250413194446/https://president.gov.by/en/gosudarstvo/prazdniki/calendar-2024>
```

PPsyrius commented on holidays/countries/brunei.py:
```diff
@@ -76,31 +76,31 @@ class Brunei(
+        * <https://web.archive.org/web/20240724020501/https://www.jpm.gov.bn/Circulars%20PDF%20Library/jpmsk10-2020.pdf>
+        * <https://web.archive.org/web/20240724020246/https://www.jpm.gov.bn/Circulars%20PDF%20Library/jpmsk10-2021.pdf>
+        * <https://web.archive.org/web/20240901140406/https://www.jpm.gov.bn/Circulars%20PDF%20Library/jpmsk04-2022.pdf>
+        * <https://web.archive.org/web/20250102120446/https://www.jpm.gov.bn/Circulars%2520PDF%2520Library/jpmsk04-2023.pdf>
```
Fixed broken link
```suggestion
        * <https://web.archive.org/web/20250428091155/https://www.jpm.gov.bn/Circulars%20PDF%20Library/jpmsk04-2023.pdf>
```

PPsyrius commented on holidays/countries/china.py:
```diff
@@ -24,38 +24,38 @@ class China(ObservedHolidayBase, ChineseCalendarHolidays, InternationalHolidays,
+        * [2009](https://web.archive.org/web/20230726083438/https://www.gov.cn/gongbao/content/2008/content_1175823.htm)
+        * [2008](https://web.archive.org/web/20240610103541/https://www.gov.cn/gongbao/content/2008/content_859870.htm)
+        * [2007](https://web.archive.org/web/20230727050141/https://www.gov.cn/gongbao/content/2007/content_503397.htm)
+        * [2006](https://web.archive.org/web/20221130214247/https://zh.wikisource.org/wiki/%E5%9B%BD%E5%8A%A1%E9%99%A2%E5%8A%9E%E5%85%AC%E5%8E%85%E5%85%B3%E4%BA%8E2006%E5%B9%B4%E9%83%A8%E5%88%86%E8%8A%82%E5%81%87%E6%97%A5%E5%AE%89%E6%8E%92%E7%9A%84%E9%80%9A%E7%9F%A5)
+        * [2005](https://web.archive.org/web/20210212184000/https://zh.wikisource.org/wiki/%E5%9B%BD%E5%8A%A1%E9%99%A2%E5%8A%9E%E5%85%AC%E5%8E%85%E5%85%B3%E4%BA%8E2005%E5%B9%B4%E9%83%A8%E5%88%86%E8%8A%82%E5%81%87%E6%97%A5%E5%AE%89%E6%8E%92%E7%9A%84%E9%80%9A%E7%9F%A5)
+        * [2004](https://web.archive.org/web/20210212183857/https://zh.wikisource.org/wiki/%E5%9B%BD%E5%8A%A1%E9%99%A2%E5%8A%9E%E5%85%AC%E5%8E%85%E5%85%B3%E4%BA%8E2004%E5%B9%B4%E9%83%A8%E5%88%86%E8%8A%82%E5%81%87%E6%97%A5%E5%AE%89%E6%8E%92%E7%9A%84%E9%80%9A%E7%9F%A5)
+        * [2003](https://web.archive.org/web/20210302090553/https://zh.wikisource.org/wiki/%E5%9B%BD%E5%8A%A1%E9%99%A2%E5%8A%9E%E5%85%AC%E5%8E%85%E5%85%B3%E4%BA%8E2003%E5%B9%B4%E9%83%A8%E5%88%86%E8%8A%82%E5%81%87%E6%97%A5%E4%BC%91%E6%81%AF%E5%AE%89%E6%8E%92%E7%9A%84%E9%80%9A%E7%9F%A5)
+        * [2002](https://web.archive.org/web/20180122201149/https://zh.wikisource.org/wiki/%E5%9B%BD%E5%8A%A1%E9%99%A2%E5%8A%9E%E5%85%AC%E5%8E%85%E5%85%B3%E4%BA%8E2002%E5%B9%B4%E9%83%A8%E5%88%86%E8%8A%82%E5%81%87%E6%97%A5%E4%BC%91%E6%81%AF%E5%AE%89%E6%8E%92%E7%9A%84%E9%80%9A%E7%9F%A5)
+        * [2001](https://web.archive.org/web/20180123032517/https://zh.wikisource.org/wiki/%E5%9B%BD%E5%8A%A1%E9%99%A2%E5%8A%9E%E5%85%AC%E5%8E%85%E5%85%B3%E4%BA%8E2001%E5%B9%B4%E6%98%A5%E8%8A%82%E3%80%81%E2%80%9C%E4%BA%94%E4%B8%80%E2%80%9D%E3%80%81%E2%80%9C%E5%8D%81%E4%B8%80%E2%80%9D%E6%94%BE%E5%81%87%E5%AE%89%E6%8E%92%E7%9A%84%E9%80%9A%E7%9F%A5)
```
```suggestion
        * [2006](https://web.archive.org/web/20221130214247/https://zh.wikisource.org/wiki/国务院办公厅关于2006年部分节假日安排的通知)
        * [2005](https://web.archive.org/web/20210212184000/https://zh.wikisource.org/wiki/国务院办公厅关于2005年部分节假日安排的通知)
        * [2004](https://web.archive.org/web/20210212183857/https://zh.wikisource.org/wiki/国务院办公厅关于2004年部分节假日安排的通知)
        * [2003](https://web.archive.org/web/20210302090553/https://zh.wikisource.org/wiki/国务院办公厅关于2003年部分节假日休息安排的通知)
        * [2002](https://web.archive.org/web/20180122201149/https://zh.wikisource.org/wiki/国务院办公厅关于2002年部分节假日休息安排的通知)
        * [2001](https://web.archive.org/web/20180123032517/https://zh.wikisource.org/wiki/国务院办公厅关于2001年春节、“五一”、“十一”放假安排的通知)
```

PPsyrius commented on holidays/countries/congo.py:
```diff
@@ -20,8 +20,8 @@ class Congo(HolidayBase, ChristianHolidays, InternationalHolidays):
-        * [Loi N° 2-94](http://mokili.free.fr/jours_feries.php)
-        * [Loi N° 18-2010](https://www.finances.gouv.cg/sites/default/files/documents/n¯18-2010%20du%2027%20novembre%202010.PDF)
+        * [Loi N° 2-94](https://web.archive.org/web/20241003070938/http://mokili.free.fr/jours_feries.php)
+        * [Loi N° 18-2010](https://web.archive.org/web/20250427173832/https://www.finances.gouv.cg/sites/default/files/documents/n%C2%AF18-2010%20du%2027%20novembre%202010.PDF)
```
```suggestion
        * [Loi N° 18-2010](https://web.archive.org/web/20250427173832/https://www.finances.gouv.cg/sites/default/files/documents/n¯18-2010%20du%2027%20novembre%202010.PDF)
```

PPsyrius commented on holidays/countries/ecuador.py:
```diff
@@ -28,7 +28,7 @@ class Ecuador(ObservedHolidayBase, ChristianHolidays, InternationalHolidays):
     References:
         * <https://en.wikipedia.org/wiki/Public_holidays_in_Ecuador>
-        * [Código del Trabajo](https://biblioteca.defensoria.gob.ec/bitstream/37000/3364/1/C%C3%B3digo%20de%20Trabajo%20%2804-11-2021%29.pdf)
+        * [Código del Trabajo](https://web.archive.org/web/20241120135529/http://biblioteca.defensoria.gob.ec/bitstream/37000/3364/1/C%C3%B3digo%20de%20Trabajo%20(04-11-2021).pdf)
```
```suggestion
        * [Código del Trabajo](https://web.archive.org/web/20250428092005/https://biblioteca.defensoria.gob.ec/bitstream/37000/3364/1/Código%20de%20Trabajo%20%2804-11-2021%29.pdf)
```

PPsyrius commented on holidays/countries/fiji.py:
```diff
@@ -37,26 +37,26 @@ class Fiji(
-         * [2023](https://web.archive.org/web/20231129154609/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
-         * [2024](https://web.archive.org/web/20250121185434/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
-         * [2025](https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
+         * [2019](https://web.archive.org/web/20191018023027/https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
+         * [2020](https://web.archive.org/web/20210103183942/https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
+         * [2021-2022](https://web.archive.org/web/20221223004409/https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
+         * [2023](https://web.archive.org/web/20231129154609/https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
+         * [2024](https://web.archive.org/web/20250121185434/https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
+         * [2025](https://web.archive.org/web/20250318092311/https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
```
```suggestion
         * [2019](https://web.archive.org/web/20191018023027/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
         * [2020](https://web.archive.org/web/20210103183942/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
         * [2021-2022](https://web.archive.org/web/20221223004409/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
         * [2023](https://web.archive.org/web/20231129154609/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
         * [2024](https://web.archive.org/web/20250121185434/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
         * [2025](https://web.archive.org/web/20250318092311/https://www.fiji.gov.fj/About-Fiji/Public-Holidays)
```

PPsyrius commented on holidays/countries/malta.py:
```diff
@@ -20,12 +20,12 @@ class Malta(HolidayBase, ChristianHolidays, InternationalHolidays):
-        * [Att 8 tal-1989 (Additional Holidays added)](https://legislation.mt/eli/act/1989/8)
-        * [Att 2 tal-2005 (If fall on weekends then not observed in terms of vacation leave)](https://legislation.mt/eli/act/2005/2/eng)
-        * [Att 4 tal-2021 (Revert Act II of 2005 changes for vacation leave)](https://legislation.mt/eli/cap/252/20210212/mlt)
+        * <https://web.archive.org/web/20250105224348/https://www.gov.mt/en/About%2520Malta/Pages/Public%2520Holidays.aspx>
```
```suggestion
        * <https://web.archive.org/web/20240406224540/https://www.gov.mt/en/About%20Malta/Pages/Public%20Holidays.aspx>
```

PPsyrius reviewed (commented):
I've checked up to letter N for now

PPsyrius commented on holidays/countries/south_korea.py:
```diff
@@ -293,9 +293,9 @@ class SouthKoreaStaticHolidays:
-        * <https://namu.wiki/w/임시공휴일> *
-        * <https://namu.wiki/w/공휴일/대한민국> **
-        * <https://namu.wiki/w/대체%20휴일%20제도>
+        * <https://web.archive.org/web/20230828094617/https://namu.wiki/w/%EC%9E%84%EC%8B%9C%EA%B3%B5%ED%9C%B4%EC%9D%BC> *
+        * <https://web.archive.org/web/20250125102245/https://namu.wiki/w/%EA%B3%B5%ED%9C%B4%EC%9D%BC/%EB%8C%80%ED%95%9C%EB%AF%BC%EA%B5%AD> **
+        * <https://web.archive.org/web/20230815034831/https://namu.wiki/w/%EB%8C%80%EC%B2%B4%20%ED%9C%B4%EC%9D%BC%20%EC%A0%9C%EB%8F%84>
```
```suggestion
        * <https://namu.wiki/w/임시공휴일> *
        * <https://namu.wiki/w/공휴일/대한민국> **
        * <https://namu.wiki/w/대체%20휴일%20제도>
```

PPsyrius commented on holidays/countries/south_korea.py:
```diff
@@ -585,7 +585,7 @@ class SouthKoreaStaticHolidays:
     }
     # Pre-2014 Alternate Holidays
-    # https://namu.wiki/w/대체%20휴일%20제도#s-4.2.1
+    # https://web.archive.org/web/20230815034831/https://namu.wiki/w/%EB%8C%80%EC%B2%B4%20%ED%9C%B4%EC%9D%BC%20%EC%A0%9C%EB%8F%84#s-4.2.1
```
```suggestion
    # https://namu.wiki/w/대체%20휴일%20제도#s-4.2.1
```

PPsyrius commented on holidays/countries/venezuela.py:
```diff
@@ -60,7 +60,7 @@ def _populate_public_holidays(self):
             self._add_holiday_apr_19(tr("Declaración de la Independencia"))
 
-        # https://bit.ly/3B4Xd1L
+        # https://web.archive.org/web/20220619063359/https://bit.ly/3b4xD1L
```
I don't know why did we use bit.ly here, but here's the web archive-ed full source
```suggestion
        # https://web.archive.org/web/20250428122616/https://venezuelaenretrospectiva.wordpress.com/2016/05/01/1o-de-mayo-dia-del-trabajador-venezolano/
```

PPsyrius commented on holidays/financial/ice_futures_europe.py:
```diff
@@ -18,10 +18,10 @@ class ICEFuturesEurope(ObservedHolidayBase, ChristianHolidays, InternationalHoli
-        * <https://web.archive.org/web/20211022183728/https://www.ice.com/publicdocs/Trading_Schedule.pdf>
+        * <https://web.archive.org/web/20241203121313/https://www.ice.com/publicdocs/futures/Trading_Schedule_Migrated_Liffe_Contracts.pdf>
+        * <https://web.archive.org/web/20241118133442/https://www.ice.com/publicdocs/Trading_Schedule.pdf>
+        * <https://web.archive.org/web/20230927015846/https://web.archive.org/web/20241118133442/https://www.ice.com/publicdocs/Trading_Schedule.pdf>
+        * <https://web.archive.org/web/20211022183728/https://web.archive.org/web/20241118133442/https://www.ice.com/publicdocs/Trading_Schedule.pdf>
```
```suggestion
        * <https://web.archive.org/web/20230927015846/https://www.ice.com/publicdocs/Trading_Schedule.pdf>
        * <https://web.archive.org/web/20211022183728/https://www.ice.com/publicdocs/Trading_Schedule.pdf>
```

PPsyrius commented on holidays/financial/ny_stock_exchange.py:
```diff
@@ -37,12 +37,12 @@ class NewYorkStockExchange(
-        * <https://s3.amazonaws.com/armstrongeconomics-wp/2013/07/NYSE-Closings.pdf>
-        * <https://web.archive.org/web/20211101162021/https://www.nyse.com/markets/hours-calendars>
+        * <https://web.archive.org/web/20221206064307/https://s3.amazonaws.com/armstrongeconomics-wp/2013/07/NYSE-Closings.pdf>
+        * <https://web.archive.org/web/20211101162021/https://web.archive.org/web/20250426144831/https://www.nyse.com/markets/hours-calendars>
```
```suggestion
* <https://web.archive.org/web/20211101162021/https://www.nyse.com/markets/hours-calendars>
```

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -46,43 +46,43 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
+        * [Constitution Petition Day renamed National Day (JUL 1938)](https://web.archive.org/web/20241109062813/https://th.wikisource.org/wiki/%E0%B8%9B%E0%B8%A3%E0%B8%B0%E0%B8%81%E0%B8%B2%E0%B8%A8%E0%B8%AA%E0%B8%B3%E0%B8%99%E0%B8%B1%E0%B8%81%E0%B8%99%E0%B8%B2%E0%B8%A2%E0%B8%81%E0%B8%A3%E0%B8%B1%E0%B8%90%E0%B8%A1%E0%B8%99%E0%B8%95%E0%B8%A3%E0%B8%B5_%E0%B8%A5%E0%B8%87%E0%B8%A7%E0%B8%B1%E0%B8%99%E ...
+        * [Holidays Act Amendment (MAR 1940)](https://web.archive.org/web/20240813142812/https://ratchakitcha.soc.go.th/documents/1110471.pdf)
+        * [Holidays Act Amendment (SEP 1940) (In-effect 1941)](https://web.archive.org/web/20231021230955/https://ratchakitcha.soc.go.th/documents/1111954.pdf)
+        * [Removal of Royal Language for King's Birthday (B.E 2484/1941)](https://web.archive.org/web/20241213001307/https://th.wikisource.org/wiki/%E0%B8%9B%E0%B8%A3%E0%B8%B0%E0%B8%81%E0%B8%B2%E0%B8%A8%E0%B8%AA%E0%B8%B3%E0%B8%99%E0%B8%B1%E0%B8%81%E0%B8%99%E0%B8%B2%E0%B8%A2%E0%B8%81%E0%B8%A3%E0%B8%B1%E0%B8%90%E0%B8%A1%E0%B8%99%E0%B8%95%E0%B8%A3%E0%B8%B5_%E0%B8%A5%E0%B8%87%E0%B8%A7%E0%B8%B1%E0%B8% ...
```
IMO `wikisource.org` should also be add to the exception like Wikipedia
```suggestion
        * [Removal of Royal Language for King's Birthday (B.E 2484/1941)](https://web.archive.org/web/20241213001307/https://th.wikisource.org/wiki/ประกาศสำนักนายกรัฐมนตรี_ลงวันที่_19_กันยายน_2484)
```

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -46,43 +46,43 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
-        * [Holidays Act Amendment (MAR 1940)](https://ratchakitcha.soc.go.th/documents/1110471.pdf)
-        * [Holidays Act Amendment (SEP 1940) (In-effect 1941)](https://ratchakitcha.soc.go.th/documents/1111954.pdf)
-        * [Removal of Royal Language for King's Birthday (B.E 2484/1941)](https://th.wikisource.org/wiki/ประกาศสำนักนายกรัฐมนตรี_ลงวันที่_19_กันยายน_2484)
+        * [Constitution Petition Day renamed National Day (JUL 1938)](https://web.archive.org/web/20241109062813/https://th.wikisource.org/wiki/%E0%B8%9B%E0%B8%A3%E0%B8%B0%E0%B8%81%E0%B8%B2%E0%B8%A8%E0%B8%AA%E0%B8%B3%E0%B8%99%E0%B8%B1%E0%B8%81%E0%B8%99%E0%B8%B2%E0%B8%A2%E0%B8%81%E0%B8%A3%E0%B8%B1%E0%B8%90%E0%B8%A1%E0%B8%99%E0%B8%95%E0%B8%A3%E0%B8%B5_%E0%B8%A5%E0%B8%87%E0%B8%A7%E0%B8%B1%E0%B8%99%E ...
```
```suggestion
        * [Constitution Petition Day renamed National Day (JUL 1938)](https://web.archive.org/web/20241109062813/https://th.wikisource.org/wiki/ประกาศสำนักนายกรัฐมนตรี_ลงวันที่_18_กรกฎาคม_2481)
```

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -46,43 +46,43 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
         * [Holidays Act, Franco-Thai War Armistice Day Repealed (B.E. 2487/1944)](https://ratchakitcha.soc.go.th/documents/1121365.pdf)
-        * [Removal of Royal Language for King's Birthday Repealed (B.E 2488/1945)](https://th.wikisource.org/wiki/ประกาศสำนักนายกรัฐมนตรี_ลงวันที่_12_มกราคม_2488)
-        * [Holidays Act Amendment (DEC B.E. 2488/1945)](https://ratchakitcha.soc.go.th/documents/1123218.pdf)
+        * [Removal of Royal Language for King's Birthday Repealed (B.E 2488/1945)](https://web.archive.org/web/20241210051421/https://th.wikisource.org/wiki/%E0%B8%9B%E0%B8%A3%E0%B8%B0%E0%B8%81%E0%B8%B2%E0%B8%A8%E0%B8%AA%E0%B8%B3%E0%B8%99%E0%B8%B1%E0%B8%81%E0%B8%99%E0%B8%B2%E0%B8%A2%E0%B8%81%E0%B8%A3%E0%B8%B1%E0%B8%90%E0%B8%A1%E0%B8%99%E0%B8%95%E0%B8%A3%E0%B8%B5_%E0%B8%A5%E0%B8%87%E0%B8%A7%E0%B8% ...
```
```suggestion
        * [Removal of Royal Language for King's Birthday Repealed (B.E 2488/1945)](https://web.archive.org/web/20241210051421/https://th.wikisource.org/wiki/ประกาศสำนักนายกรัฐมนตรี_ลงวันที่_12_มกราคม_2488)
```

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -46,43 +46,43 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
+        * [Holidays Act Amendment (MAR 1940)](https://web.archive.org/web/20240813142812/https://ratchakitcha.soc.go.th/documents/1110471.pdf)
+        * [Holidays Act Amendment (SEP 1940) (In-effect 1941)](https://web.archive.org/web/20231021230955/https://ratchakitcha.soc.go.th/documents/1111954.pdf)
+        * [Removal of Royal Language for King's Birthday (B.E 2484/1941)](https://web.archive.org/web/20241213001307/https://th.wikisource.org/wiki/%E0%B8%9B%E0%B8%A3%E0%B8%B0%E0%B8%81%E0%B8%B2%E0%B8%A8%E0%B8%AA%E0%B8%B3%E0%B8%99%E0%B8%B1%E0%B8%81%E0%B8%99%E0%B8%B2%E0%B8%A2%E0%B8%81%E0%B8%A3%E0%B8%B1%E0%B8%90%E0%B8%A1%E0%B8%99%E0%B8%95%E0%B8%A3%E0%B8%B5_%E0%B8%A5%E0%B8%87%E0%B8%A7%E0%B8%B1%E0%B8% ...
         * [Holidays Act, Franco-Thai War Armistice Day added (B.E. 2485/1942)](https://ratchakitcha.soc.go.th/documents/1114825.pdf)
         * [Holidays Act, Franco-Thai War Armistice Day Repealed (B.E. 2487/1944)](https://ratchakitcha.soc.go.th/documents/1121365.pdf)
```
The same goes for other ratchakitcha.soc.go.th sources
```suggestion
        * [Holidays Act, Franco-Thai War Armistice Day added (B.E. 2485/1942)](https://web.archive.org/web/20250428123610/https://ratchakitcha.soc.go.th/documents/1114825.pdf)
        * [Holidays Act, Franco-Thai War Armistice Day Repealed (B.E. 2487/1944)](https://web.archive.org/web/20250428123653/https://ratchakitcha.soc.go.th/documents/1121365.pdf)
```

PPsyrius replied:
Nvm, I'll sort Thai ones myself later alongside tiny.cc cleanup in a suggestion PR

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -95,54 +95,54 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
+            * <https://web.archive.org/web/20250126073943/https://hilight.kapook.com/view/148862>
+        * [National Father's Day](https://web.archive.org/web/20230205075650/https://www.brh.go.th/index.php/2019-02-27-04-12-21/594-5-5)
         * Constitution Day:
             * [ประเทศไทย](https://th.wikipedia.org/wiki/วันรัฐธรรมนูญ_)
```
```suggestion
            * <https://th.wikipedia.org/wiki/วันรัฐธรรมนูญ>
```

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -95,54 +95,54 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
+            * <https://web.archive.org/web/20250212210436/https://hilight.kapook.com/view/18208>
             * [Bank of Thailand](http://tiny.cc/wa_bot_1992)
-            * <https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2475.aspx>
+            * <https://web.archive.org/web/20161028001043/http://www.myhora.com:80/%E0%B8%9B%E0%B8%8F%E0%B8%B4%E0%B8%97%E0%B8%B4%E0%B8%99/%E0%B8%9B%E0%B8%8F%E0%B8%B4%E0%B8%97%E0%B8%B4%E0%B8%99-%E0%B8%9E.%E0%B8%A8.2475.aspx>
```
```suggestion
            * <https://web.archive.org/web/20161028001043/http://www.myhora.com:80/ปฏิทิน/ปฏิทิน-พ.ศ.2475.aspx>
```

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -95,54 +95,54 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
-            * <https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2475.aspx>
+            * <https://web.archive.org/web/20161028001043/http://www.myhora.com:80/%E0%B8%9B%E0%B8%8F%E0%B8%B4%E0%B8%97%E0%B8%B4%E0%B8%99/%E0%B8%9B%E0%B8%8F%E0%B8%B4%E0%B8%97%E0%B8%B4%E0%B8%99-%E0%B8%9E.%E0%B8%A8.2475.aspx>
         * New Year's Eve:
             * [Bank of Thailand](http://tiny.cc/wa_bot_1992)
```
All tiny.cc sources are just shortened Internet Archive links, AFAIK I did used them for Thailand, Canada, and South Africa back when I first joined to prevent links from going over character width limit - will need to double check them later
```suggestion
            * <https://web.archive.org/web/20230205072056/https%3A//www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx>
```

PPsyrius committed:
Remove tiny.cc source link aliases, Thailand sources archive work

kritibirda committed:
Apply suggestions from code review

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda committed:
Update holidays/countries/south_korea.py

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda26 commented on holidays/countries/argentina.py:
```diff
@@ -42,107 +42,107 @@ class Argentina(
+        * <https://web.archive.org/web/20250219140544/https://web.archive.org/web/20250213230843/https://www.argentina.gob.ar/interior/feriados-nacionales-2022>
+        * <https://web.archive.org/web/20250219140544/https://web.archive.org/web/20250320070326/https://www.argentina.gob.ar/interior/feriados-nacionales-2023>
+        * <https://web.archive.org/web/20250219140544/https://web.archive.org/web/20250405100504/https://www.argentina.gob.ar/interior/feriados-nacionales-2024>
+        * <https://web.archive.org/web/20250219140544/https://web.archive.org/web/20250426102342/https://www.argentina.gob.ar/interior/feriados-nacionales-2025>
```
I figured the error happens when the same link is present more than once in a file, once archived and another time unarchived.

kritibirda committed:
Merge branch 'archive-links' into remove_tiny_cc

Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda committed:
Merge pull request #1 from PPsyrius/remove_tiny_cc

Remove tiny.cc source link aliases, Thailand sources archive work

kritibirda committed:
replace tinyurl.com links for barbados and namibia

kritibirda committed:
archive more links

kritibirda committed:
Merge branch 'dev' into archive-links

KJhellico commented on holidays/countries/south_africa.py:
```diff
@@ -152,6 +152,6 @@ class SouthAfricaStaticHolidays:
 
     special_public_holidays_observed = {
-        # Special holiday http://tiny.cc/za_y2k
+        # https://web.archive.org/web/20120328122217/http%3A//www.info.gov.za/speeches/1999/991028409p1002.htm
```
```suggestion
        # https://web.archive.org/web/20120328122217/http://www.info.gov.za/speeches/1999/991028409p1002.htm
```

KJhellico commented on holidays/countries/thailand.py:
```diff
@@ -46,103 +46,103 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
 
     Checked with:
-        * [Bank of Thailand lists](http://tiny.cc/wa_bot_2023)
+        * [Bank of Thailand](https://web.archive.org/web/20230205072056/https%3A//www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx)
```
```suggestion
        * [Bank of Thailand](https://web.archive.org/web/20230205072056/https://www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx)
```

KJhellico replied:
```suggestion
        * [Bank of Thailand](https://web.archive.org/web/20230205072056/https://www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx)
```

KJhellico commented on holidays/countries/thailand.py:
```diff
@@ -46,103 +46,103 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
+        * [Chakri Memorial Day](https://web.archive.org/web/20221003075521/https://www.ocac.go.th/news/๖-เมษายน-วันจักรี/v)
         * Songkran Festival:
-            * [museumsiam.org](http://tiny.cc/wa_museumsiam_songkran)
+            * [museumsiam.org](https://web.archive.org/web/20230205074402/https%3A//m.museumsiam.org/da-detail2.php?MID=3&CID=177&CONID=4033)
```
```suggestion
            * [museumsiam.org](https://web.archive.org/web/20230205074402/https://m.museumsiam.org/da-detail2.php?MID=3&CID=177&CONID=4033)
```

KJhellico commented on holidays/countries/thailand.py:
```diff
@@ -46,103 +46,103 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
-        * [National Labour Day](https://www.thairath.co.th/lifestyle/culture/1832869)
-        * [National Day (24 June: Defunct)](http://tiny.cc/wa_mincul_nat_day)
+        * [National Labour Day](https://web.archive.org/web/20220629085454/https://www.thairath.co.th/lifestyle/culture/1832869)
+        * [National Day (24 June: Defunct)](https://web.archive.org/web/20091106202525/http%3A//www.culture.go.th/study.php?&YY=2548&MM=11&DD=2)
```
```suggestion
        * [National Day (24 June: Defunct)](https://web.archive.org/web/20091106202525/http://www.culture.go.th/study.php?&YY=2548&MM=11&DD=2)
```

KJhellico commented on holidays/countries/thailand.py:
```diff
@@ -46,103 +46,103 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
-            * [Ministry of Culture](http://tiny.cc/wa_mincul_nat_day)
-            * <https://hilight.kapook.com/view/148862>
-        * [National Father's Day](https://www.brh.go.th/index.php/2019-02-27-04-12-21/594-5-5)
+            * [Ministry of Culture](https://web.archive.org/web/20091106202525/http%3A//www.culture.go.th/study.php?&YY=2548&MM=11&DD=2)
```
```suggestion
            * [Ministry of Culture](https://web.archive.org/web/20091106202525/http://www.culture.go.th/study.php?&YY=2548&MM=11&DD=2)
```

KJhellico commented on holidays/countries/thailand.py:
```diff
@@ -46,103 +46,103 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
-            * <https://www.myhora.com/ปฏิทิน/ปฏิทิน-พ.ศ.2475.aspx>
+            * <https://th.wikipedia.org/wiki/วันรัฐธรรมนูญ>
+            * <https://web.archive.org/web/20250212210436/https://hilight.kapook.com/view/18208>
+            * [Bank of Thailand](https://web.archive.org/web/20230205072056/https%3A//www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx)
```
```suggestion
            * [Bank of Thailand](https://web.archive.org/web/20230205072056/https://www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx)
```

KJhellico commented on holidays/countries/thailand.py:
```diff
@@ -46,103 +46,103 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
-        * [Visakha Bucha](https://www.onab.go.th/th/content/category/detail/id/73/iid/3401)
-        * [Asarnha Bucha](https://www.onab.go.th/th/content/category/detail/id/73/iid/3397)
-        * [Buddhist Lent Day](https://www.onab.go.th/th/content/category/detail/id/73/iid/3395)
+            * [Bank of Thailand](https://web.archive.org/web/20230205072056/https%3A//www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx)
```
```suggestion
            * [Bank of Thailand](https://web.archive.org/web/20230205072056/https://www.bot.or.th/Thai/FinancialInstitutions/FIholiday/Pages/2023.aspx)
```

KJhellico commented on holidays/countries/singapore.py:
```diff
@@ -45,9 +45,9 @@ class Singapore(
     References:
         * [Wikipedia](https://en.wikipedia.org/wiki/Public_holidays_in_Singapore)
-        * [Holidays Act](https://sso.agc.gov.sg/Act/HA1998) (Act 24 of
+        * [Holidays Act](https://web.archive.org/web/20250405061431/http://sso.agc.gov.sg/Act/HA1998) (Act 24 of
             1968—Holidays (Amendment) Act 1968)
```
```suggestion
        * [Holidays Act 1998](https://web.archive.org/web/20250405061431/https://sso.agc.gov.sg/Act/HA1998)
```

KJhellico commented on holidays/financial/ny_stock_exchange.py:
```diff
@@ -37,12 +37,12 @@ class NewYorkStockExchange(
-        * <https://www.nyse.com/publicdocs/nyse/regulation/nyse/NYSE_Rules.pdf>
-        * <https://www.nyse.com/markets/hours-calendars>
+        * <https://web.archive.org/web/20240118104341/https://www.nyse.com/publicdocs/nyse/regulation/nyse/NYSE_Rules.pdf>
+        * <https://web.archive.org/web/20250426144831/https://www.nyse.com/markets/hours-calendars>
```
```suggestion
```

PPsyrius replied:
>         * <https://web.archive.org/web/20240118104341/https://www.nyse.com/publicdocs/nyse/regulation/nyse/NYSE_Rules.pdf>

and

>         * <https://web.archive.org/web/20250426144831/https://www.nyse.com/markets/hours-calendars>

are two different links AFAIK

KJhellico replied:
We have https://www.nyse.com/markets/hours-calendars in "Historical data" below.

KJhellico commented on holidays/financial/ny_stock_exchange.py:
```diff
@@ -37,12 +37,12 @@ class NewYorkStockExchange(
-        * <https://s3.amazonaws.com/armstrongeconomics-wp/2013/07/NYSE-Closings.pdf>
-        * <https://web.archive.org/web/20211101162021/https://www.nyse.com/markets/hours-calendars>
+        * <https://web.archive.org/web/20221206064307/https://s3.amazonaws.com/armstrongeconomics-wp/2013/07/NYSE-Closings.pdf>
+* <https://web.archive.org/web/20211101162021/https://www.nyse.com/markets/hours-calendars>
```
```suggestion
        * <https://web.archive.org/web/20211101162021/https://www.nyse.com/markets/hours-calendars>
```

PPsyrius commented on holidays/countries/kazakhstan.py:
```diff
@@ -31,9 +31,9 @@ class Kazakhstan(
+        * <https://web.archive.org/web/20240918145208/https://adilet.zan.kz/kaz/docs/Z990000493_>
 
     Islamic holidays:
         * [2025](https://en.inform.kz/news/first-day-of-ramadan-to-fall-on-march-1-2025-ca393f/)
```
```suggestion
        * [2025](https://web.archive.org/web/20250429084936/https://qazinform.com/news/first-day-of-ramadan-to-fall-on-march-1-2025-ca393f)
```

PPsyrius commented on holidays/countries/hongkong.py:
```diff
@@ -59,11 +59,11 @@ class HongKong(
+        * [Holidays for 2010-2025](https://web.archive.org/web/20250402133857/https://www.labour.gov.hk/eng/news/holidays_list.htm)
 
     General Holidays:
         * [Cap. 149 General Holidays Ordinance](https://www.elegislation.gov.hk/hk/cap149!en-zh-Hant-HK?INDEX_CS=N)
```
```suggestion
        * [Cap. 149 General Holidays Ordinance](http://archive.today/2025.04.29-125900/https://www.elegislation.gov.hk/hk/cap149!en-zh-Hant-HK?INDEX_CS=N)
```

PPsyrius commented on holidays/countries/united_states.py:
```diff
@@ -36,51 +36,51 @@ class UnitedStates(ObservedHolidayBase, ChristianHolidays, InternationalHolidays
-            * <https://governor.gov.mp/archived-news/executive-actions-archive/memorandum-2022-legal-holidays/>
-            * <https://webcache.googleusercontent.com/search?q=cache:C17_7FBgPtQJ:https://governor.gov.mp/archived-news/executive-actions-archive/memorandum-2022-legal-holidays/&hl=en&gl=sg&strip=1&vwsrc=0>
+            * <https://web.archive.org/web/20240727045236/https://governor.gov.mp/archived-news/executive-actions-archive/memorandum-2022-legal-holidays/>
+            * <https://webcache.googleusercontent.com/search?q=cache:C17_7FBgPtQJ:https://web.archive.org/web/20240727045236/https://governor.gov.mp/archived-news/executive-actions-archive/memorandum-2022-legal-holidays/&hl=en&gl=sg&strip=1&vwsrc=0>
```
```suggestion
            * <https://web.archive.org/web/20250429140158/https://governor.gov.mp/archived-news/executive-actions-archive/memorandum-2022-legal-holidays/>
```

PPsyrius commented on holidays/countries/jersey.py:
```diff
@@ -162,22 +162,22 @@ class JerseyStaticHolidays:
+        * <https://web.archive.org/web/20250418060211/https://www.gov.je/News/2021/pages/cornriots.aspx>
+        * <https://web.archive.org/web/20250414072751/https://www.jerseylaw.je/laws/enacted/Pages/Jersey%20R%20%20O%209288.aspx>
+        * <https://web.archive.org/web/20250414072814/https://www.jerseylaw.je/laws/enacted/Pages/Jersey%20R%20%20O%209317.aspx>
         * <https://www.jerseylaw.je/laws/enacted/Pages/Jersey%20RO%2042-2001.aspx>
```
```suggestion
        * <https://web.archive.org/web/20250429110054/https://www.jerseylaw.je/laws/enacted/Pages/RO-042-2001.aspx>
```

PPsyrius commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,344 @@
+                    all_unique_urls.update(
+                        url
+                        for url in urls_in_file
+                        if url and (url.startswith("http://") or url.startswith("https://"))
```
```suggestion
                        if url
                        and (url.startswith("http://") or url.startswith("https://"))  # NOSONAR
```

PPsyrius commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,344 @@
+        allowed_methods=["HEAD", "GET", "OPTIONS", "POST"],
+    )
+    adapter = HTTPAdapter(max_retries=retries)
+    session.mount("http://", adapter)
```
```suggestion
    session.mount("http://", adapter)  # NOSONAR
```

PPsyrius reviewed (commented):
I did look into SonarQube's error suppression a bit, seems like they only got `# NOSONAR` as in-line global issue suppression tool with no option to only disable specific rule like `mypy` or other tools...

https://stackoverflow.com/questions/61717134/sonarqube-how-to-supress-a-specific-warning-in-python-using-nosonar

kritibirda committed:
Apply suggestions from code review

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Co-authored-by: ~Jhellico <KJhellico@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda committed:
Apply suggestions from code review

Co-authored-by: ~Jhellico <KJhellico@users.noreply.github.com>
Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda committed:
Apply suggestions from code review

Co-authored-by: ~Jhellico <KJhellico@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda committed:
Merge branch 'dev' into archive-links

Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

Roshan-1024 committed:
Add Trinidad and Tobago holidays (#2402)

Signed-off-by: Roshan Pradhan <roshanpradhan1024@gmail.com>
Co-authored-by: ~Jhellico <jhellico@gmail.com>

github-actions[bot] committed:
chore: Update snapshots (#2510)

Co-authored-by: github-merge-queue[bot] <118344674+github-merge-queue[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/download-artifact from 4.2.1 to 4.3.0 (#2512)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump actions/setup-python from 5.5.0 to 5.6.0 (#2513)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump cyclonedx-bom from 5.5.0 to 6.0.0 (#2514)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump ruff from 0.11.6 to 0.11.7 (#2515)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump importlib-metadata from 8.6.1 to 8.7.0 (#2516)

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

PPsyrius commented on holidays/countries/taiwan.py:
```diff
@@ -54,20 +54,20 @@ class Taiwan(ObservedHolidayBase, ChineseCalendarHolidays, InternationalHolidays
 
     Commemorative Day and Day Implementation Method Amendments:
         * [Ministry of Interior (87) Order No. 8706459](https://zh.wikisource.org/wiki/紀念日及節日實施辦法_(民國87年))
```
```suggestion
        * [Ministry of Interior (87) Order No. 8706459](https://web.archive.org/web/20250501035936/https://zh.wikisource.org/wiki/紀念日及節日實施辦法_(民國87年))
```

PPsyrius commented on holidays/countries/thailand.py:
```diff
@@ -46,105 +46,105 @@ class Thailand(ObservedHolidayBase, InternationalHolidays, StaticHolidays, ThaiC
         * Songkran Festival:
-            * [museumsiam.org](http://tiny.cc/wa_museumsiam_songkran)
+            * [museumsiam.org](https://web.archive.org/web/20230205074402/https://m.museumsiam.org/da-detail2.php?MID=3&CID=177&CONID=4033)
             * <https://resolution.soc.go.th/?prep_id=123659>
```
```suggestion
            * <https://web.archive.org/web/20250501040114/https://resolution.soc.go.th/?prep_id=123659>
```

PPsyrius commented on holidays/countries/ukraine.py:
```diff
@@ -169,53 +169,53 @@ class UkraineStaticHolidays:
+        * [2001 [1]](https://web.archive.org/web/20220312201133/https://zakon.rada.gov.ua/laws/show/138-2001-%D1%80/)
+        * [2001 [2]](https://web.archive.org/web/20220404230657/https://zakon.rada.gov.ua/laws/show/210-2001-%D0%BF)
+        * [2002](https://web.archive.org/web/20220521085829/https://zakon.rada.gov.ua/laws/show/202-2002-%D1%80)
         * [2002 - 2003](https:/zakon.rada.gov.ua/laws/show/705-2002-%D1%80)
```
```suggestion
        * [2002 - 2003](https://archive.ph/2025.05.01-040415/https://zakon.rada.gov.ua/laws/show/705-2002-р%23Text)
```

KJhellico replied:
Previous (non latest) WA link work too: https://web.archive.org/web/20220312195735/https://zakon.rada.gov.ua/laws/show/705-2002-р

PPsyrius reviewed (commented):
These are probably the last few archivable unarchived reference links out there - I'll see if we can further optimize our `scripts/archive_links.py` 👍

KJhellico commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,348 @@
+        )
+        response.raise_for_status()
+        data = response.json()
+        if len(data) > 1:
+            # The first row is the header, the rest are captures
+            latest = data[1]
+            timestamp = latest[1]
+            original = latest[2]
+            wayback_url = f"https://web.archive.org/web/{timestamp}/{original}"
+            return wayback_url
```
Latest capture may be unusable, better to query more and check for statuscode - [4]

kritibirda26 replied:
The CDX allows filtering on the statuscode, we can choose a list of statuscodes to allow or disallow etc? What do you think of a filter on `statuscode:200`?

KJhellico replied:
Yes, seems 200 will be optimal choice.

kritibirda26 replied:
Done.

kritibirda committed:
Apply suggestions from code review

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

KJhellico commented on holidays/countries/ukraine.py:
```diff
@@ -169,53 +169,53 @@ class UkraineStaticHolidays:
+        * [2001 [1]](https://web.archive.org/web/20220312201133/https://zakon.rada.gov.ua/laws/show/138-2001-%D1%80/)
+        * [2001 [2]](https://web.archive.org/web/20220404230657/https://zakon.rada.gov.ua/laws/show/210-2001-%D0%BF)
+        * [2002](https://web.archive.org/web/20220521085829/https://zakon.rada.gov.ua/laws/show/202-2002-%D1%80)
+        * [2002 - 2003](https://archive.ph/2025.05.01-040415/https://zakon.rada.gov.ua/laws/show/705-2002-р%23Text)
```
```suggestion
        * [2002 - 2003](https://web.archive.org/web/20220312195735/https://zakon.rada.gov.ua/laws/show/705-2002-%D1%80)
```

KJhellico commented on holidays/countries/uruguay.py:
```diff
@@ -28,10 +28,10 @@ class Uruguay(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, Sta
+        * [Decree Law #9000](https://web.archive.org/web/20231029150729/https://www.impo.com.uy/bases/decretos-ley/9000-1933)
+        * [Decree Law #14977](https://web.archive.org/web/20231013033006/https://www.impo.com.uy/bases/decretos-ley/14977-1979)
+        * [Decree Law #15535](https://web.archive.org/web/20210519002815/https://www.impo.com.uy/bases/decretos-ley/15535-1984/)
         * [Law #16805](http://www.parlamento.gub.uy/leyes/AccesoTextoLey.asp?Ley=16805)
         * [Law #17414](http://www.parlamento.gub.uy/leyes/AccesoTextoLey.asp?Ley=17414)
```
```suggestion
        * [Law #16805](https://web.archive.org/web/20241121064154/https://parlamento.gub.uy/documentosyleyes/leyes/ley/16805)
        * [Law #17414](https://web.archive.org/web/20241121064131/https://parlamento.gub.uy/documentosyleyes/leyes/ley/17414)
```

KJhellico commented on holidays/countries/tanzania.py:
```diff
@@ -33,21 +33,21 @@ class Tanzania(
         * [1962](https://old.tanzlii.org/tz/legislation/act/1962/48/)
         * [1964](https://old.tanzlii.org/tz/legislation/act/1964/52/)
         * [1966](https://old.tanzlii.org/tz/legislation/act/1966/39/)
```
```suggestion
        * [1964](https://web.archive.org/web/20250426030228/http://www.parliament.go.tz/polis/uploads/bills/acts/1565172859-The%20Public%20Holidays%20Ordinance%20(amendment)%20Act,%201964.pdf)
        * [1966](https://web.archive.org/web/20241130122429/http://www.parliament.go.tz/polis/uploads/bills/acts/1565692654-The%20Public%20Holidays%20Ordinance%20(Amendment)%20Act,%201966.pdf)
```

kritibirda committed:
Apply suggestions from code review

Co-authored-by: ~Jhellico <KJhellico@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

KJhellico committed:
Fix `TestClosestHoliday` current date handling (#2517)

kritibirda committed:
archive more links

kritibirda committed:
add filter for captures with only 200 status code

arkid15r committed:
Disable v1 incompatibility warning (#2518)

github-actions[bot] committed:
Update pre-commit hooks (#2519)

Co-authored-by: arkid15r <2201626+arkid15r@users.noreply.github.com>

PPsyrius commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,349 @@
+        print(f"Warning: Could not read file {filepath} due to: {e}", file=sys.stderr)
+        return []
+
+    found_urls = set()
+    try:
+        for match in re.findall(FIND_URL_PATTERN, file_content):
+            if match:
+                ignore = False
+                for ignore_pattern in IGNORE_URL_PATTERNS:
+                    if re.search(ignore_pattern, match):
+                        ignore = True
+                        break
+                if not ignore:
+                    cleaned_match = match.rstrip(")")
+                    found_urls.add(cleaned_match)
+        return sorted(list(found_urls))
```
We managed to reduce the initial check time from 7.54373020 to 1.87061540 seconds by switching Python's `any( )` 🎉 
```suggestion
    try:
        found_urls = {
            match.rstrip(")")
            for match in re.findall(FIND_URL_PATTERN, file_content)
            if match and not any(p.search(match) for p in IGNORE_URL_PATTERNS)
        }
        return sorted(found_urls)
```

kritibirda committed:
Apply suggestions from code review

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

KJhellico commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,349 @@
+    re.compile(r"loc\.gov"),
+    re.compile(r"iso\.org"),
+    re.compile(r"semver\.org"),
+]
```
```
r"archive\.org"
...
r"web\.archive\.org"
...
```
`r"archive\.org"` is enough.

PPsyrius replied:
AFAIK it got added to prevent double Internet Archive links, but from a quick test a few hours ago the issue still persists regardless if both the unarchive and archive version of the same link exists in the same file.

kritibirda26 replied:
Hi! Yes, I forgot to fix that issue. :sweat:

KJhellico commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,344 @@
+    try:
+        found_urls = {
+            match.rstrip(")")
+            for match in re.findall(FIND_URL_PATTERN, file_content)
```
```suggestion
            for match in FIND_URL_PATTERN.findall(file_content)
```

KJhellico commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,344 @@
+from urllib3.util.retry import Retry
+
+EXTENSIONS_TO_SCAN = [".py", ".po"]
+FIND_URL_PATTERN = re.compile(r'https?://[^\s<>"]+|www\.[^\s<>"]+')
```
In fact, we have no links that don't start with http(s), do we can simplify this regex and no check with `url.startswith` below.

kritibirda26 commented on scripts/archive_links.py:
```diff
@@ -0,0 +1,344 @@
+    parser = argparse.ArgumentParser(
+        description="Find, check/archive via Wayback Machine, and replace URLs in project files.",
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        "--archive-policy",
+        choices=["if-missing", "always", "never"],
+        default="if-missing",
+        help="Archiving policy: 'if-missing' (only archive if no capture exists), "
+        "'always' (always submit for archiving), 'never' (only search for "
+        "exising captures)",
+    )
+    args = parser.parse_args()
```
Can you tell how the links existence check is different from the only check in WM? The only check in WM is the never mode behavior IIUC.

KJhellico replied:
"links existence check" - checking the existence of original links (not archived)
"only check in WM" - checking the presence of links in WM (what is "never" policy does now)

kritibirda26 replied:
Added the links existence check mode. I think the `--if-missing` policy covers the only links search and display mode. Is there any other mode that should be added?

KJhellico replied:
I would call this mode "show-only". 

> "links existence check" - checking the existence of original links (not archived)

I meant the request (HTTP HEAD) of the original links to check their liveness.

kritibirda26 replied:
Thanks for clarifying. I'll update the script.

kritibirda26 replied:
Added a check-liveness mode.

KJhellico committed:
Fix typography: replace U+2019 with "'" and U+2013 with '-' (#2523)

KJhellico committed:
Add missing subdivisions aliases  (#2520)

Signed-off-by: ~Jhellico <KJhellico@users.noreply.github.com>
Co-authored-by: Arkadii Yakovets <2201626+arkid15r@users.noreply.github.com>

arkid15r committed:
Finalize v0.72

arkid15r committed:
Initialize v0.73

PPsyrius committed:
Update Argentina holidays: add 2018 G20 Leaders' Summit for Buenos Aires (#2529)

kritibirda committed:
address changes from code review

1. add check only mode
2. handle cases where the same url is present in a file in both archived and unarchived form.

kritibirda committed:
Merge remote-tracking branch 'origin/dev' into archive-links

# Conflicts:
#	holidays/countries/albania.py
#	holidays/countries/hongkong.py
#	holidays/countries/macau.py

kritibirda committed:
address changes from code review - 2

kritibirda committed:
simplify valid url check

kritibirda26 commented:
Apologies, I rebased the branch instead of merging and that led to this commit mess. Please let me know if this is fine for now or I should try to fix it? I'll be more careful in future.

KJhellico commented:
> Please let me know if this is fine for now or if I should try to fix it?

Now it's seems ok for me (part of commits have a different hash from the identical ones in dev, but anyway they will squash on merging). Just don't do that again. :)

KJhellico commented:
BTW, I checked out the script while working on #2530. The experience didn't go exactly as I had expected. On the first run, the script replaced only two links that were already in WM and one just added to WM (while adding all new ones to WM):
```
Processing file: D:\@Work\test_holidays\test\holidays\andorra.py
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=GD_2024_10_24_14_36_16
        Archive submission complete (status: 200)
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QAH20191128_08_58_08
        Archive submission complete (status: 200)
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QAH20201210_11_13_19
        Archive submission complete (status: 200)
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QAH20211111_16_06_47
        Archive submission complete (status: 200)
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QAH20221117_10_31_45
        Archive submission complete (status: 200)
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QAH_2023_11_17_12_40_42
        Archive submission complete (status: 200)
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QAH_2024_11_21_09_44_47
        Archive submission complete (status: 200)...
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QCH_2024_10_10_11_42_15
        Archive submission complete (status: 200)
        Using existing capture for https://www.bopa.ad/Documents/Detall?doc=QEH_2023_12_15_13_03_32 (archive_policy=if-missing).
        Using existing capture for https://www.bopa.ad/Documents/Detall?doc=QEH_2024_11_27_10_29_49 (archive_policy=if-missing).
        Submitting for archive: https://www.bopa.ad/Documents/Detall?doc=QMH20191209_09_08_10
        Archive submission complete (status: 200)
...
  Attempting replacements in: D:\@Work\test_holidays\test\holidays\andorra.py
    Replaced 'https://www.bopa.ad/Documents/Detall?doc=QAH_2023_11_17_12_40_42' (1x)
    Replaced 'https://www.bopa.ad/Documents/Detall?doc=QEH_2023_12_15_13_03_32' (1x)
    Replaced 'https://www.bopa.ad/Documents/Detall?doc=QEH_2024_11_27_10_29_49' (1x)
  Successfully wrote 3 replacement(s) to D:\@Work\test_holidays\test\holidays\andorra.py
```
On the second run, it replaced all the rest:
```
Processing file: D:\@Work\test_holidays\test\holidays\andorra.py
        Using existing capture for https://www.bopa.ad/Documents/Detall?doc=GD_2024_10_24_14_36_16 (archive_policy=if-missing).
        Using existing capture for https://www.bopa.ad/Documents/Detall?doc=QAH20191128_08_58_08 (archive_policy=if-missing).
        Using existing capture for https://www.bopa.ad/Documents/Detall?doc=QAH20201210_11_13_19 (archive_policy=if-missing).
...
  Attempting replacements in: D:\@Work\test_holidays\test\holidays\andorra.py
    Replaced 'https://www.bopa.ad/Documents/Detall?doc=GD_2024_10_24_14_36_16' (1x)
    Replaced 'https://www.bopa.ad/Documents/Detall?doc=QAH20191128_08_58_08' (1x)
    Replaced 'https://www.bopa.ad/Documents/Detall?doc=QAH20201210_11_13_19' (1x)
    Replaced 'https://www.bopa.ad/Documents/Detall?doc=QAH20211111_16_06_47' (1x)
...
  Successfully wrote 37 replacement(s) to D:\@Work\test_holidays\test\holidays\andorra.py
```

kritibirda26 commented:
The replacement issue is likely because the WM does not index URLs immediately, at least not always in the span of the HTTP request queuing the url. So usually a second run is needed a few minutes later to fetch and update the rest of the urls. I have been using the script by running it in `if-missing` mode first and then again in `never` mode.

I can update the checking for the WM URLs at the end of the submission to loop over and over for multiple attempts or some minutes in case the more URLs become available. What do you think?

KJhellico commented:
> The replacement issue is likely because the WM does not index URLs immediately, at least not always in the span of the HTTP request queuing the url. So usually a second run is needed a few minutes later to fetch and update the rest of the urls.

I also assumed that this was the reason.

> I can update the checking for the WM URLs at the end of the submission to loop over and over for multiple attempts or some minutes in case the more URLs become available. What do you think?

I think this will be an overcomplication. A second run will be enough.

kritibirda committed:
Refactor URL processing logic and add liveness check mode

Modularized the script by splitting functionalities into dedicated functions for readability and maintainability. Added "--check-liveness" mode to validate if URLs are alive with HTTP HEAD requests. Updated argument parsing to handle mutually exclusive modes more cleanly.

kritibirda committed:
Merge branch 'dev' into archive-links

kritibirda26 commented:
Hi again, Sorry for the delay in updating the PR, I was busy with academics.

kritibirda committed:
Merge branch 'dev' into archive-links

kritibirda committed:
archive more links

KJhellico commented on .pre-commit-config.yaml:
```diff
@@ -52,6 +52,8 @@ repos:
         additional_dependencies:
           - types-polib
           - types-python-dateutil
+          - types-setuptools
```
```suggestion
```
This appeared incorrectly on dev merge.

PPsyrius commented on holidays/countries/argentina.py:
```diff
@@ -813,20 +813,23 @@ class ArgentinaStaticHolidays:
-        * [Decree 42/2022](https://www.argentina.gob.ar/normativa/nacional/decreto-42-2022-360018/texto)
-        * [Decree 842/2022](https://www.argentina.gob.ar/normativa/nacional/decreto-842-2022-376857/texto)
+        * [Decree 615/2010](https://web.archive.org/web/20240109181152/https://www.argentina.gob.ar/normativa/nacional/decreto-615-2010-166825/texto)
+        * [Joint Resolution 1159/2010](https://web.archive.org/web/20231215002151/https://www.argentina.gob.ar/normativa/nacional/resoluci%C3%B3n-1159-2010-173748/texto)
```
```suggestion
        * [Joint Resolution 1159/2010](https://web.archive.org/web/20231215002151/https://www.argentina.gob.ar/normativa/nacional/resolución-1159-2010-173748/texto)
```

PPsyrius commented on holidays/countries/panama.py:
```diff
@@ -23,11 +23,11 @@ class Panama(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, Stat
-        * [Law #55 of Nov 7, 2001](https://docs.panama.justia.com/federales/leyes/55-de-2001-nov-14-2001.pdf)
+        * [Labor Code 1947](https://web.archive.org/web/20250429081529/https://docs.panama.justia.com/federales/leyes/67-de-1947-nov-26-1947.pdf)
+        * [Cabinet Decree #347 of 1969](https://web.archive.org/web/20250429081519/https://docs.panama.justia.com/federales/decretos-de-gabinete/decreto-de-gabinete-347-de-1969-nov-14-1969.pdf)
+        * [Labor Code 1971](https://web.archive.org/web/20240420050417/https://www.mitradel.gob.pa/wp-content/uploads/2016/12/c%C3%B3digo-detrabajo.pdf)
```
```suggestion
        * [Labor Code 1971](https://web.archive.org/web/20240420050417/https://www.mitradel.gob.pa/wp-content/uploads/2016/12/código-detrabajo.pdf)
```

PPsyrius commented on holidays/countries/ukraine.py:
```diff
@@ -169,53 +169,53 @@ class UkraineStaticHolidays:
+        * [1993 [2]](https://web.archive.org/web/20220501192004/https://zakon.rada.gov.ua/laws/show/725-92-%D0%BF/)
+        * [1994](https://web.archive.org/web/20220423134711/https://zakon.rada.gov.ua/laws/show/98-94-%D0%BF)
+        * [1995 [1]](https://web.archive.org/web/20220416193351/https://zakon.rada.gov.ua/laws/show/852-95-%D0%BF/)
+        * [1995 [2]](https://web.archive.org/web/20220727233924/https://zakon.rada.gov.ua/laws/show/634-95-%D0%BF)
+        * [1995 [3]](https://web.archive.org/web/20220404230852/https://zakon.rada.gov.ua/laws/show/266-95-%D0%BF)
+        * [1996](https://web.archive.org/web/20220703182454/https://zakon.rada.gov.ua/laws/show/424-96-%D0%BF)
+        * [1997 [1]](https://web.archive.org/web/20220710133208/https://zakon.rada.gov.ua/laws/show/326-97-%D0%BF)
+        * [1997 [2]](https://web.archive.org/web/20240707083032/https://zakon.rada.gov.ua/laws/show/1547-96-%D0%BF)
+        * [1998](https://web.archive.org/web/20220516171244/https://zakon.rada.gov.ua/laws/show/1404-97-%D0%BF>)
+        * [1999 [1]](https://web.archive.org/web/20220721004702/https://zakon.rada.gov.ua/laws/show/1433-99-%D0%BF>)
+        * [1999 [2]](https://web.archive.org/web/20220701225902/https://zakon.rada.gov.ua/laws/show/558-99-%D0%BF>)
+        * [1999 [3]](https://web.archive.org/web/20220703131420/https://zakon.rada.gov.ua/laws/show/2070-98-%D0%BF)
+        * [2000 [1]](https://web.archive.org/web/20220416193413/https://zakon.rada.gov.ua/laws/show/1251-2000-%D0%BF/)
+        * [2000 [2]](https://web.archive.org/web/20220404231224/https://zakon.rada.gov.ua/laws/show/717-2000-%D0%BF)
+        * [2001 [1]](https://web.archive.org/web/20220312201133/https://zakon.rada.gov.ua/laws/show/138-2001-%D1%80/)
+        * [2001 [2]](https://web.archive.org/web/20220404230657/https://zakon.rada.gov.ua/laws/show/210-2001-%D0%BF)
+        * [2002](https://web.archive.org/web/20220521085829/https://zakon.rada.gov.ua/laws/show/202-2002-%D1%80)
+        * [2002 - 2003](https://web.archive.org/web/20220312195735/https://zakon.rada.gov.ua/laws/show/705-2002-%D1%80)
+        * [2004](https://web.archive.org/web/20220404105708/https://zakon.rada.gov.ua/laws/show/773-2003-%D1%80)
+        * [2005 [1]](https://web.archive.org/web/20220521235321/https://zakon.rada.gov.ua/laws/show/936-2004-%D1%80)
+        * [2005 [2]](https://web.archive.org/web/20220611030516/https://zakon.rada.gov.ua/laws/show/133-2005-%D1%80)
+        * [2006 [1]](https://web.archive.org/web/20240822140051/https://zakon.rada.gov.ua/laws/show/490-2005-%D1%80)
+        * [2006 [2]](https://web.archive.org/web/20220312195751/https://zakon.rada.gov.ua/laws/show/562-2005-%D1%80/)
+        * [2007](https://web.archive.org/web/20240823064327/https://zakon.rada.gov.ua/laws/show/612-2006-%D1%80)
+        * [2008 [1]](https://web.archive.org/web/20240823064327/https://zakon.rada.gov.ua/laws/show/1059-2007-%D1%80)
+        * [2008 [2]](https://web.archive.org/web/20240901160821/https://zakon.rada.gov.ua/laws/show/538-2008-%D1%80)
+        * [2009](https://web.archive.org/web/20220312195619/https://zakon.rada.gov.ua/laws/show/1458-2008-%D1%80/)
+        * [2010](https://web.archive.org/web/20240826001002/https://zakon.rada.gov.ua/laws/show/1412-2009-%D1%80)
+        * [2011](https://web.archive.org/web/20220312200622/https://zakon.rada.gov.ua/laws/show/2130-2010-%D1%80/)
+        * [2012](https://web.archive.org/web/20250119122439/https://zakon.rada.gov.ua/laws/show/1210-2011-%D1%80)
+        * [2013](https://web.archive.org/web/20250119201324/https://zakon.rada.gov.ua/laws/show/1043-2012-%D1%80)
+        * [2014](https://web.archive.org/web/20250421100048/https://zakon.rada.gov.ua/laws/show/920-2013-%D1%80)
+        * [2015](https://web.archive.org/web/20240801225558/https://zakon.rada.gov.ua/laws/show/1084-2014-%D1%80)
+        * [2016](https://web.archive.org/web/20221210054440/https://zakon.rada.gov.ua/laws/show/1155-2015-%D1%80)
+        * [2017](https://web.archive.org/web/20240609025258/https://zakon.rada.gov.ua/laws/show/850-2016-%D1%80)
+        * [2018](https://web.archive.org/web/20221210060148/https://zakon.rada.gov.ua/laws/show/1-2018-%D1%80)
+        * [2019](https://web.archive.org/web/20220316200919/https://zakon.rada.gov.ua/laws/show/7-2019-%D1%80)
+        * [2020](https://web.archive.org/web/20250423064733/https://zakon.rada.gov.ua/laws/show/995-2019-%D1%80)
+        * [2021](https://web.archive.org/web/20250402142530/https://zakon.rada.gov.ua/laws/show/1191-2020-%D1%80)
+        * [2022](https://web.archive.org/web/20250404010912/https://zakon.rada.gov.ua/laws/show/1004-2021-%D1%80)
```
```suggestion
        * [1991](https://web.archive.org/web/20220830105426/https://zakon.rada.gov.ua/laws/show/60-91-п)
        * [1992 [1]](https://web.archive.org/web/20220816132241/https://zakon.rada.gov.ua/laws/show/202-92-п)
        * [1992 [2]](https://web.archive.org/web/20220514124422/https://zakon.rada.gov.ua/laws/show/377-91-п)
        * [1993 [1]](https://web.archive.org/web/20220429231922/https://zakon.rada.gov.ua/laws/show/563-93-п/)
        * [1993 [2]](https://web.archive.org/web/20220501192004/https://zakon.rada.gov.ua/laws/show/725-92-п/)
        * [1994](https://web.archive.org/web/20220423134711/https://zakon.rada.gov.ua/laws/show/98-94-п)
        * [1995 [1]](https://web.archive.org/web/20220416193351/https://zakon.rada.gov.ua/laws/show/852-95-п/)
        * [1995 [2]](https://web.archive.org/web/20220727233924/https://zakon.rada.gov.ua/laws/show/634-95-п)
        * [1995 [3]](https://web.archive.org/web/20220404230852/https://zakon.rada.gov.ua/laws/show/266-95-п)
        * [1996](https://web.archive.org/web/20220703182454/https://zakon.rada.gov.ua/laws/show/424-96-п)
        * [1997 [1]](https://web.archive.org/web/20220710133208/https://zakon.rada.gov.ua/laws/show/326-97-п)
        * [1997 [2]](https://web.archive.org/web/20240707083032/https://zakon.rada.gov.ua/laws/show/1547-96-п)
        * [1998](https://web.archive.org/web/20220516171244/https://zakon.rada.gov.ua/laws/show/1404-97-п>)
        * [1999 [1]](https://web.archive.org/web/20220721004702/https://zakon.rada.gov.ua/laws/show/1433-99-п>)
        * [1999 [2]](https://web.archive.org/web/20220701225902/https://zakon.rada.gov.ua/laws/show/558-99-п>)
        * [1999 [3]](https://web.archive.org/web/20220703131420/https://zakon.rada.gov.ua/laws/show/2070-98-п)
        * [2000 [1]](https://web.archive.org/web/20220416193413/https://zakon.rada.gov.ua/laws/show/1251-2000-п/)
        * [2000 [2]](https://web.archive.org/web/20220404231224/https://zakon.rada.gov.ua/laws/show/717-2000-п)
        * [2001 [1]](https://web.archive.org/web/20220312201133/https://zakon.rada.gov.ua/laws/show/138-2001-р/)
        * [2001 [2]](https://web.archive.org/web/20220404230657/https://zakon.rada.gov.ua/laws/show/210-2001-п)
        * [2002](https://web.archive.org/web/20220521085829/https://zakon.rada.gov.ua/laws/show/202-2002-р)
        * [2002 - 2003](https://web.archive.org/web/20220312195735/https://zakon.rada.gov.ua/laws/show/705-2002-р)
        * [2004](https://web.archive.org/web/20220404105708/https://zakon.rada.gov.ua/laws/show/773-2003-р)
        * [2005 [1]](https://web.archive.org/web/20220521235321/https://zakon.rada.gov.ua/laws/show/936-2004-р)
        * [2005 [2]](https://web.archive.org/web/20220611030516/https://zakon.rada.gov.ua/laws/show/133-2005-р)
        * [2006 [1]](https://web.archive.org/web/20240822140051/https://zakon.rada.gov.ua/laws/show/490-2005-р)
        * [2006 [2]](https://web.archive.org/web/20220312195751/https://zakon.rada.gov.ua/laws/show/562-2005-р/)
        * [2007](https://web.archive.org/web/20240823064327/https://zakon.rada.gov.ua/laws/show/612-2006-р)
        * [2008 [1]](https://web.archive.org/web/20240823064327/https://zakon.rada.gov.ua/laws/show/1059-2007-р)
        * [2008 [2]](https://web.archive.org/web/20240901160821/https://zakon.rada.gov.ua/laws/show/538-2008-р)
        * [2009](https://web.archive.org/web/20220312195619/https://zakon.rada.gov.ua/laws/show/1458-2008-р/)
        * [2010](https://web.archive.org/web/20240826001002/https://zakon.rada.gov.ua/laws/show/1412-2009-р)
        * [2011](https://web.archive.org/web/20220312200622/https://zakon.rada.gov.ua/laws/show/2130-2010-р/)
        * [2012](https://web.archive.org/web/20250119122439/https://zakon.rada.gov.ua/laws/show/1210-2011-р)
        * [2013](https://web.archive.org/web/20250119201324/https://zakon.rada.gov.ua/laws/show/1043-2012-р)
        * [2014](https://web.archive.org/web/20250421100048/https://zakon.rada.gov.ua/laws/show/920-2013-р)
        * [2015](https://web.archive.org/web/20240801225558/https://zakon.rada.gov.ua/laws/show/1084-2014-р)
        * [2016](https://web.archive.org/web/20221210054440/https://zakon.rada.gov.ua/laws/show/1155-2015-р)
        * [2017](https://web.archive.org/web/20240609025258/https://zakon.rada.gov.ua/laws/show/850-2016-р)
        * [2018](https://web.archive.org/web/20221210060148/https://zakon.rada.gov.ua/laws/show/1-2018-р)
        * [2019](https://web.archive.org/web/20220316200919/https://zakon.rada.gov.ua/laws/show/7-2019-р)
        * [2020](https://web.archive.org/web/20250423064733/https://zakon.rada.gov.ua/laws/show/995-2019-р)
        * [2021](https://web.archive.org/web/20250402142530/https://zakon.rada.gov.ua/laws/show/1191-2020-р)
        * [2022](https://web.archive.org/web/20250404010912/https://zakon.rada.gov.ua/laws/show/1004-2021-р)
```

PPsyrius commented on holidays/countries/venezuela.py:
```diff
@@ -20,13 +20,13 @@ class Venezuela(HolidayBase, ChristianHolidays, InternationalHolidays):
+        * <https://web.archive.org/web/20250317124608/https://dias-festivos.eu/dias-festivos/venezuela/>
+        * [1909 Aug 5](https://web.archive.org/web/20241012151721/https://www.guao.org/sites/default/files/efemerides/69.Ley%20fiestas%20nacionales%201909.pdf)
+        * [1918 May 19](https://web.archive.org/web/20250429081541/https://www.guao.org/sites/default/files/efemerides/70.%20Ley%20de%20fiestas%20nacionales%201918.pdf)
+        * [1921 Jun 11](https://web.archive.org/web/20241012160001/https://www.guao.org/sites/default/files/efemerides/37.LEYES_Y_DECRETOS_1921_Di%CC%81a_de_la_raza.PDF)
```
```suggestion
        * [1921 Jun 11](https://web.archive.org/web/20241012160001/https://www.guao.org/sites/default/files/efemerides/37.LEYES_Y_DECRETOS_1921_Día_de_la_raza.PDF)
```

PPsyrius reviewed (commented):
Additional ones by @KJhellico from #2580

kritibirda committed:
Apply suggestions from code review

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda committed:
add requests to dev dependencies, remove wrongly added types-setuptools

kritibirda committed:
Merge branch 'dev' into archive-links

PPsyrius commented:
I think this is already pretty LGTM once this is updated with the latest changes - what do you think? @KJhellico @arkid15r

kritibirda committed:
Merge branch 'dev' into archive-links

KJhellico committed:
Clean up unicode URLs

KJhellico reviewed (dismissed):
LGTM!

PPsyrius commented on holidays/countries/canada.py:
```diff
@@ -33,14 +33,14 @@ class Canada(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, Stat
+        * <https://web.archive.org/web/20250427131243/https://web2.gov.mb.ca/laws/statutes/ccsm/r120e.php>
+        * <https://web.archive.org/web/20250112160822/https://www2.gnb.ca/content/gnb/en/departments/elg/local_government/content/governance/content/days_of_rest_act.html>
+        * <https://web.archive.org/web/20250405170509/https://www.ontario.ca/document/your-guide-employment-standards-act-0/public-holidays>
         * <https://www.gov.nu.ca/en/print/pdf/node/52266>
```
```suggestion
        * <https://archive.org/details/nunavut-day-designated-as-a-general-holiday-start-date>
```

kritibirda committed:
Apply suggestions from code review

Co-authored-by: Panpakorn Siripanich <19505219+PPsyrius@users.noreply.github.com>
Signed-off-by: Kriti Birda <164247895+kritibirda26@users.noreply.github.com>

kritibirda committed:
Merge branch 'dev' into archive-links

kritibirda committed:
Archive more links and add some documents uploaded manually to Internet archive

KJhellico reviewed (dismissed):
LGTM! 👍

arkid15r reviewed (commented):
@kritibirda26 great work -- both idea and implementation 👏 

I don't want to be a blocker here just because of readability and best practices suggestions. Moreover, refactoring w/o tests is a bit tricky. So I'll just provide a general feedback you might use later:
 - if you need just a sequence prefer using tuples instead of lists
 - use consistent naming (ignore vs ignored)
 - use Path instead of os.path
 - if dict (or other) params order doesn't matter -- order alphabetically
 - use spellcheck locally (e.g. your IDE plugin)

arkid15r committed:
Update scripts/archive_links.py

Signed-off-by: Arkadii Yakovets <2201626+arkid15r@users.noreply.github.com>
